File: patch_casedetail.py
import re

with open('src/components/CaseDetailModal.tsx', 'r') as f:
    content = f.read()

# Add imports for User, Building, MapPin if not present
if 'import { EntitiesView }' not in content:
    content = content.replace("import { EvidenceArchiveView }", "import { EntitiesView } from './EntitiesView';\nimport { EvidenceArchiveView }")
    content = content.replace("Scale, Database,", "Scale, Database, User, Building, MapPin,")

# Add the tabs to the nav strip. Let's place them before Rabbit Hole (tab 7).
nav_strip_patch = """
          <button
            onClick={() => { setActiveTab('people'); sound.click(); }}
            className={`px-3.5 py-2.5 text-xs font-mono font-bold whitespace-nowrap transition-all border-b-2 flex items-center gap-1.5 ${
              activeTab === 'people'
                ? 'border-cyan-400 text-cyan-300 bg-cyan-950/20'
                : 'border-transparent text-gray-400 hover:text-gray-200'
            }`}
          >
            <User className="w-3.5 h-3.5" />
            <span className="hidden sm:inline">People</span>
          </button>
          
          <button
            onClick={() => { setActiveTab('organisations'); sound.click(); }}
            className={`px-3.5 py-2.5 text-xs font-mono font-bold whitespace-nowrap transition-all border-b-2 flex items-center gap-1.5 ${
              activeTab === 'organisations'
                ? 'border-cyan-400 text-cyan-300 bg-cyan-950/20'
                : 'border-transparent text-gray-400 hover:text-gray-200'
            }`}
          >
            <Building className="w-3.5 h-3.5" />
            <span className="hidden sm:inline">Organisations</span>
          </button>
          
          <button
            onClick={() => { setActiveTab('locations'); sound.click(); }}
            className={`px-3.5 py-2.5 text-xs font-mono font-bold whitespace-nowrap transition-all border-b-2 flex items-center gap-1.5 ${
              activeTab === 'locations'
                ? 'border-cyan-400 text-cyan-300 bg-cyan-950/20'
                : 'border-transparent text-gray-400 hover:text-gray-200'
            }`}
          >
            <MapPin className="w-3.5 h-3.5" />
            <span className="hidden sm:inline">Locations</span>
          </button>
"""

# Insert right before Rabbit Hole tab
content = content.replace(
    "<button\n            onClick={() => { setActiveTab('rabbithole');", 
    nav_strip_patch + "\n          <button\n            onClick={() => { setActiveTab('rabbithole');"
)

# Render the tabs
tabs_render_patch = """
          {/* TAB: PEOPLE */}
          {activeTab === 'people' && (
            <EntitiesView caseFileId={currentCase.id} type="people" currentUser={currentUser} />
          )}
          
          {/* TAB: ORGANISATIONS */}
          {activeTab === 'organisations' && (
            <EntitiesView caseFileId={currentCase.id} type="organisations" currentUser={currentUser} />
          )}
          
          {/* TAB: LOCATIONS */}
          {activeTab === 'locations' && (
            <EntitiesView caseFileId={currentCase.id} type="locations" currentUser={currentUser} />
          )}
"""

# Insert right before Rabbit Hole content
content = content.replace(
    "{/* TAB 7: RABBIT HOLE CONNECTIONS */}",
    tabs_render_patch + "\n          {/* TAB 7: RABBIT HOLE CONNECTIONS */}"
)

with open('src/components/CaseDetailModal.tsx', 'w') as f:
    f.write(content)

File: patch_case_detail.js
import fs from 'fs';
let content = fs.readFileSync('src/components/CaseDetailModal.tsx', 'utf8');

if (!content.includes("TimelineView")) {
  content = content.replace(
    "import { X, Calendar, Lock, Globe, Users, ShieldAlert, Folder, Download, ExternalLink, Hash, Clock, FileText, ChevronRight, Eye, RefreshCw, MessageSquare, Plus, CheckCircle, Search, Save, AlertTriangle, Trash2, BrainCircuit, Activity, Edit3, Image as ImageIcon, Scale, FolderArchive, ArrowRight, CornerRightDown, MapPin, Building2, User } from 'lucide-react';",
    "import { X, Calendar, Lock, Globe, Users, ShieldAlert, Folder, Download, ExternalLink, Hash, Clock, FileText, ChevronRight, Eye, RefreshCw, MessageSquare, Plus, CheckCircle, Search, Save, AlertTriangle, Trash2, BrainCircuit, Activity, Edit3, Image as ImageIcon, Scale, FolderArchive, ArrowRight, CornerRightDown, MapPin, Building2, User } from 'lucide-react';\nimport { TimelineView } from './TimelineView';"
  );
  
  // Find where timeline tab is rendered.
  // We want to add `<TimelineView entityType="case_files" entityId={caseId} />` below the legacy JSON timeline.
  
  content = content.replace(
    /\{\(currentCase\.timeline \|\| \[\]\)\.map\(\(t, idx\) => \([\s\S]*?\}\)/,
    "{(currentCase.timeline || []).map((t: any, idx: number) => (\n                      <div key={idx} className=\"relative flex gap-6 pb-8 last:pb-0\">\n                        <div className=\"flex flex-col items-center\">\n                          <div className=\"w-3 h-3 rounded-full bg-cyan-500 shadow-[0_0_10px_rgba(6,182,212,0.6)] z-10\" />\n                          <div className=\"w-px h-full bg-gray-800 -mt-1\" />\n                        </div>\n                        <div className=\"pt-0\">\n                          <span className=\"text-cyan-400 font-mono text-sm tracking-widest font-bold\">\n                            {t.date}\n                          </span>\n                          <h4 className=\"text-gray-200 mt-1 uppercase text-sm tracking-wider font-semibold\">\n                            {t.event}\n                          </h4>\n                          <p className=\"text-gray-400 text-sm mt-2 max-w-2xl\">\n                            {t.description}\n                          </p>\n                        </div>\n                      </div>\n                    ))}\n                  </div>\n                </div>\n              </div>\n              <div className=\"mt-8\">\n                <TimelineView entityType=\"case_files\" entityId={caseId} />"
  );

  fs.writeFileSync('src/components/CaseDetailModal.tsx', content);
}
